CMakeLists.txt

```diff
@@ -111,10 +112,9 @@ endif ()
 
 
 # Set the default namespace
-if (OSL_NAMESPACE)
-    add_definitions ("-DOSL_NAMESPACE=${OSL_NAMESPACE}")
-endif ()
-message(STATUS "Setting Namespace to: ${OSL_NAMESPACE}")
+set (PROJ_NAMESPACE "${${PROJ_NAME}_NAMESPACE}")
```

Comment:
Right, there were two commits that were part of this PR.

I did backport the change that hides the osl_* symbols that were `extern "C"` for LLVM's benefit, which were never properly hidden or namespaced before. (Though I'm nervous about changing the default to be hidden in the release branches; for 1.7/1.8, you'll want to explicitly build with HIDE_SYMBOLS=1.)

I am intentionally not backporting the changes to C++ namespaces for the things that were already namespace-protected. That sounded like a riskier thing to change for a release branch, and as well, the changes to support it involve some more widely-diverged sections of the cmake scripts that would be harder to backport correctly without errors.